Cyrillic in requests

math-88 opened the pull request:
For what we used `url.PathUnescape` in client.do?
For cyrillic we must encode characters.

groovili commented:
First of all - if you need the support of cyrillic symbols - add it.
`url.PathUnescape` is used to avoid manual symbols replacing in URL (looks odd) and make it human readable for debugging. 
Why did you change .mod file? Don't see tests for cyrillic queries as well.

math-88 commented:
For using cyrillic symbols need to remove `url.PathUnescape` method. It destroys the exchange data, this will decode the string.

Also, I think you used `url.PathUnescape` because without it the request caused an error.
Because you used incorrect time format "today+12-m", you need to use "today 12-m". Without `+`.
`+` used only in Encoded string.
For example in native google request:
{"comparisonItem":[{"keyword":"kia","geo":"US","time":"now 7-d"},{"keyword":"hyundai","geo":"US","time":"now 7-d"}],"category":0,"property":""}

Added successfull tests.

Sorry, I changed .mod file because I needed to use quickly my fork in real project. Merge request should not have these changes.I fixed it.

groovili commented:
@math-88 thanks for your contribution. With your changes cyrillic symbols would be available for all package methods, but parameter changes will break backward compatibility.
At the moment some people already use this package in their projects.
If you want this change to be merged you should provide support for both cases - 
```today+12-m``` and ```today 12-m```.

math-88 commented:
Ok, let's add a hook to support invalid time values.
I made it at the Explore function. At the start point.

Tests old+new is DONE.

groovili commented:
@math-88 ok, thanks.
